### ISSUE TITLE
Undefined symbols for architecture x86_64:   "_OBJC_CLASS_$_AWSCognitoIdentityProviderASF"

### ISSUE THREAD

**Issue by vimitahirpara** [aws-amplify/aws-sdk-ios#3036]
Getting error while using AWSMobileClient pod in application 

Showing All Messages
**Ld /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSMobileClient/AWSMobileClient.framework/AWSMobileClient normal (in target 'AWSMobileClient' from project 'Pods')
    cd /Users/maya_somase/Downloads/dht-mobile-wrapper-TARDIS_Auth_API_Impl/ClinicalWrapperApp_iOS/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target x86_64-apple-ios9.0-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSMobileClient -F/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSMobileClient -F/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSAuthCore -F/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSCognitoIdentityProvider -F/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSCognitoIdentityProviderASF -F/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSCore -filelist /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/AWSMobileClient.build/Objects-normal/x86_64/AWSMobileClient.LinkFileList -install_name @rpath/AWSMobileClient.framework/AWSMobileClient -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/AWSMobileClient.build/Objects-normal/x86_64/AWSMobileClient_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/AWSMobileClient.build/Objects-normal/x86_64/AWSMobileClient.swiftmodule -framework AWSAuthCore -framework AWSCognitoIdentityProvider -framework Foundation -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/AWSMobileClient.build/Objects-normal/x86_64/AWSMobileClient_dependency_info.dat -o /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/AWSMobileClient/AWSMobileClient.framework/AWSMobileClient

Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_AWSCognitoIdentityProviderASF", referenced from:
      objc-class-ref in AWSCognitoAuth.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)**

I had used AWSMobileClient pod in my framework project.
After that I created .**xcframework** from it and integrated that in an application. I had added AWSMobileClient pod in application as well. But getting above error while building or running app on simulator. It builds successfully for device.

This was working in Xocde 11

Currently using Xcode version 12.0
iOS 14.0
AWSMobileClient is latest 2.16.0

**Comment by ruiguoamz** [comment]
Hi, @vimitahirpara 
Thanks for reaching out. 

This is an issue we are also seeing and investigating.

**Comment by vimitahirpara** [comment]
I get xcode 12 recommended updates warning in my app to change the pods deployment target to 12.0.
If I apply those changes then it sets deployment target to 12.0 to all pods target. 
If I build application it builds successfully but if I run it on simulator it crashes with error 

dyld: Symbol not found: _$s15AWSMobileClient0aB5ErrorO017idTokenAndAcceessE9NotIssuedyACSS_tcACmFWC
  Referenced from: /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/CoreSDK.framework/CoreSDK
  Expected in: /Users/maya_somase/Library/Developer/CoreSimulator/Devices/1577F9E5-35BF-45CC-8876-8D76C982D81E/data/Containers/Bundle/Application/9E308C34-C725-4CFD-AC1A-2CEC1883315E/ClinicalWrapper.app/Frameworks/AWSMobileClient.framework/AWSMobileClient
 in /Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator/CoreSDK.framework/CoreSDK
dyld: launch, loading dependent libraries
DYLD_SHARED_CACHE_DIR=/Users/maya_somase/Library/Developer/CoreSimulator/Caches/dyld/19G73/com.apple.CoreSimulator.SimRuntime.iOS-14-0.18A372
DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot
DYLD_LIBRARY_PATH=/Users/maya_somase/Library/Developer/Xcode/DerivedData/ClinicalWrapper-ghlyjxzrixwurjeruzoucyzdsrlu/Build/Products/Debug-iphonesimulator:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libBacktraceRecording.dylib:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/De
(lldb)

**Comment by ruiguoamz** [comment]
Hi, @vimitahirpara 
Can you try setting `BUILD_LIBRARIES_FOR_DISTRIBUTION ` to `YES` in your project's `Build Settings` and see if the error go away? I tried out this way in my laptop and it works

**Comment by vimitahirpara** [comment]
Hi @ruiguoamz 

I tried to set BUILD_LIBRARIES_FOR_DISTRIBUTION to YES in the application where I had integrated the .xcframework and used that pod but still getting same runtime error.

**Comment by julien-tamade** [comment]
I'm also experiencing this. Any fix?

**Comment by palpatim** [comment]
We are actively working on this and are currently testing a PR. We'll report back when we have more info.

**Comment by palpatim** [comment]
Hi all, this should be fixed with the [2.17.0](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.17.0) release. If you pin to a particular version of the SDK, please update your Podfiles to point to at least `~> 2.17.0`.

Please note that this release includes a couple of breaking changes, one directly related to supporting Xcode 12 (we no longer ship the `libAWSCognitoIdentityProviderASFBinary.a` static library. This won't affect most customers, since most projects don't directly depend on the **AWSCognitoIdentityProviderASF** framework. However, if you do depend directly on **AWSCognitoIdentityProviderASF**, make sure you remove any project references to that static library and link to the dynamic framework instead.

Please let us know if you have any questions.

**Comment by palpatim** [comment]
See #2927

**Comment by royjit** [comment]
Closing since there is no activity. Feel free to reach back if you face any issue.

**Comment by mhrjnamar** [comment]
Hi 
I am getting this error while build app for iOS
'''Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$__TtC20amplify_auth_cognito16SwiftAuthCognito", referenced from:
      objc-class-ref in AuthCognito.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)'''

I have tried these.
This error seems to be caused by amplify SDK I am using for AWS.
So I had implemented amplify SDK to a new project where it works fine.
